src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt

```diff
@@ -19,54 +23,43 @@ import uk.gov.justice.digital.hmpps.managerecallsapi.domain.PrisonId
-
-    val updatedRecall = authenticatedClient.updateRecall(recallId, UpdateRecallRequest(contraband = true))
-
-    val auditList = authenticatedClient.auditForField(recallId, "contraband")
-
-    assertThat(auditList.size, equalTo(1))
-    assertThat(auditList[0].updatedValue, equalTo(true))
-    assertThat(auditList[0].recallId, equalTo(recallId))
-    assertThat(auditList[0].updatedByUserName, equalTo(FullName("Bertie Badger")))
-    assertOffsetDateTimesEqual(auditList[0].updatedDateTime, updatedRecall.lastUpdatedDateTime)
+  private fun auditHistoryFieldValues(): Stream<Arguments>? {
+    val today = LocalDate.now()
+    return Stream.of(
+      Arguments.of(FieldName("currentPrison"), "String or Enum", UpdateRecallRequest(currentPrison = PrisonId("ABC")), "ABC"),
+      Arguments.of(FieldName("contraband"), "Boolean", UpdateRecallRequest(contraband = true), true),
+      Arguments.of(FieldName("lastReleaseDate"), "LocalDate", UpdateRecallRequest(lastReleaseDate = today), today.toString()),
+      Arguments.of(FieldName("assessedByUserId"), "UUID", UpdateRecallRequest(assessedByUserId = authenticatedClient.userId), authenticatedClient.userId.toString()),
+      Arguments.of(FieldName("sentencingInfo.sentenceLength.sentenceYears"), "nested Integer", UpdateRecallRequest(sentenceLength = Api.SentenceLength(3, 0, 0), sentenceDate = today, licenceExpiryDate = today, indexOffence = "Offence 1", sentencingCourt = CourtId("ACCRYC"), sentenceExpiryDate = today), 3),
+      Arguments.of(FieldName("sentencingInfo.sentenceExpiryDate"), "nested LocalDate", UpdateRecallRequest(sentenceLength = Api.SentenceLength(3, 0, 0), sentenceDate = today, licenceExpiryDate = today, indexOffence = "Offence 1", sentencingCourt = CourtId("ACCRYC"), sentenceExpiryDate = today), today.toString()),
+    )
   }
 
-  @Test
-  fun `get lastReleaseDate (LocalDate value) audit history`() {
-    val savedRecall = authenticatedClient.bookRecall(BookRecallRequest(nomsNumber, FirstName("Brian"), null, LastName("Badgering")))
+  @ParameterizedTest(name = "get {0} ({1} value) audit history")
+  @MethodSource("auditHistoryFieldValues")
+  fun `get audit history for single field`(
+    fieldName: FieldName,
+    fieldType: String,
+    updateRequest: UpdateRecallRequest,
+    expectedValue: Any
+  ) {
+    val savedRecall =
+      authenticatedClient.bookRecall(BookRecallRequest(nomsNumber, FirstName("Brian"), null, LastName("Badgering")))
     val recallId = savedRecall.recallId
 
-    val updatedRecall = authenticatedClient.updateRecall(recallId, UpdateRecallRequest(lastReleaseDate = LocalDate.now()))
+    val updatedRecall = authenticatedClient.updateRecall(recallId, updateRequest)
 
-    val auditList = authenticatedClient.auditForField(recallId, "lastReleaseDate")
+    val auditList = authenticatedClient.auditForField(recallId, fieldName.value)
 
     assertThat(auditList.size, equalTo(1))
-    assertThat(auditList[0].updatedValue, equalTo(LocalDate.now().toString()))
```

Comment:
FYI I quite like to see one non-parameterized test - which can be more complete/specific/explicit - alongside a parameterized test for breadth of coverage.